RIOT-OS/RIOT
# boards/imx8mq: add imx8 m4 support for IMX8MQEVK board

### Contribution description
This PR is a WIP proof of concept to add support for the CM4 of IMX8MQ cpu.
This CPU contains a quad A53 + one cortex M4F for low-power application and this PR adds supports ONLY FOR CM4. Linux or Android should run on the A53 part.
The CM4 have access to the same IP as the A53 so users must ensure that peripherals are not use by A53 and M4 at the same time. Some mechanisms to prevent such behavior exiss but are not present in this PR but this can be add in a followup PR.

Users should allow be aware that CM4 has no dedicated ROM/Flash. So applications won't restart automatically after a shutdown. However, it is possible to use U-boot or Linux to load RIOT firmware in RAM and start CM4.
Regarding RAM, the CM4 uses a small portion of the LPPDR4.

By using JLink, you should be able to load your firmware on RAM (LPDDR4) but this require to have LPDDR4 initialized which is done by the U-boot SPL.

Having RIOT running on the CM4 allow user to remotely wakeup their Linux/Android or retrieve sensors data without to have a whole Linux system running at 1Ghz for instance.
For now, only GPIOs, UART and Timer drivers should work. 

Reviews, comments and tests are highly appreciated.

Datasheet can be access for free but required login into nxp website.
[IMX8MQ link](https://www.nxp.com/design/development-boards/i.mx-evaluation-and-development-boards/evaluation-kit-for-the-i.mx-8m-applications-processor:MCIMX8M-EVK)

@kaspar030 and @emmanuelsearch may be interested by this PR.


### Testing procedure
Attach a JLink probe to the IMX8MQEVK board, connect a microUSB cable on the debug USB. Two serial ports should pop-up, the first is dedicated to Linux/Android on the A53 and the second is used by RIOT. Run all kind of tests.


### Issues/PRs references
None.
